typed/buffer_test.go

```diff
@@ -81,27 +81,44 @@ func TestReadWrite(t *testing.T) {
 	w.WriteLen8String("hello")
 	w.WriteLen16String("This is a much larger string")
 	require.NoError(t, w.Err())
+}
 
-	var b bytes.Buffer
-	w.FlushTo(&b)
-
-	r := NewReadBufferWithSize(1024)
-	r.FillFrom(bytes.NewReader(b.Bytes()), len(b.Bytes()))
-
-	assert.Equal(t, uint64(0x0123456789ABCDEF), r.ReadUint64())
-	assert.Equal(t, uint32(0xABCDEF01), r.ReadUint32())
-	assert.Equal(t, uint16(0x2345), r.ReadUint16())
-	assert.Equal(t, uint64(1), r.ReadUvarint())
-	assert.Equal(t, uint64(math.MaxInt16), r.ReadUvarint())
-	assert.Equal(t, uint64(math.MaxInt32), r.ReadUvarint())
-	assert.Equal(t, uint64(math.MaxInt64), r.ReadUvarint())
-	assert.Equal(t, byte(0xFF), r.ReadSingleByte())
-	assert.Equal(t, s, r.ReadString(len(s)))
-	assert.Equal(t, bslice, r.ReadBytes(len(bslice)))
-	assert.Equal(t, "hello", r.ReadLen8String())
-	assert.Equal(t, "This is a much larger string", r.ReadLen16String())
-
-	require.NoError(t, r.Err())
+func TestReadBufferTracking(t *testing.T) {
+	bs := []byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9}
+
+	// Run twice, once on the original buffer, and once after wrapping.
+	rbuf := NewReadBuffer(bs)
+	for i := 0; i < 2; i++ {
+		if i == 1 {
+			rbuf.Wrap(bs)
+		}
```

Comment:
would be easier to follow to iterate over two `*ReadBuffer`s, e.g. `for _, r := range []*Readbuffer{...}` checking indices in a loop can get a bit confusing

Reply:
yeah, agree, this is a little ugly. However, I want to use the same read buffer wrapped (to make sure that `initialLength` is reset). I've changed the loop to instead be over whether to wrap or not, let me know what you think of the new approach.